File: backend/src/main/java/com/programmers/pcquotation/domain/admin/service/AdminService.java
package com.programmers.pcquotation.domain.admin.service;

import java.util.Optional;

import org.springframework.stereotype.Service;

import com.programmers.pcquotation.domain.admin.entitiy.Admin;
import com.programmers.pcquotation.domain.admin.repository.AdminRepository;
import com.programmers.pcquotation.domain.member.entitiy.Member;
import com.programmers.pcquotation.domain.seller.entitiy.Seller;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class AdminService {
	private final AdminRepository adminRepository;
	public Optional<Admin> findAdminByUsername(String username) {
		return adminRepository.findByUsername(username);
	}
	public Optional<Member> findById(Long id) {
		return adminRepository.findById(id).map(admin -> admin);
	}

	public void create(Admin admin){
		adminRepository.save(admin);
	}
	public Optional<Member> findByApiKey(String apiKey) {
		return adminRepository.findByApiKey(apiKey).map(admin-> admin);
	}
}

File: backend/src/main/java/com/programmers/pcquotation/domain/admin/service/AdminService.kt
package com.programmers.pcquotation.domain.admin.service

import com.programmers.pcquotation.domain.admin.entity.Admin
import com.programmers.pcquotation.domain.admin.repository.AdminRepository
import com.programmers.pcquotation.domain.member.entity.Member
import com.programmers.pcquotation.domain.member.exception.UserNotFoundException
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class AdminService(
    private val adminRepository: AdminRepository
) {

    fun findAdminByUsername(username: String): Admin? {
        return adminRepository.findByUsername(username)
    }

    fun findById(id: Long): Member {
        return adminRepository.findByIdOrNull(id)
            ?: throw UserNotFoundException("username does not exist")
    }

    fun create(admin: Admin) {
        adminRepository.save(admin)
    }

    fun findByApiKey(apiKey: String): Member {
        return adminRepository.findByApiKey(apiKey)
            ?: throw UserNotFoundException("username does not exist")
    }
}